### PR TITLE
feat: add fetch historical feature values notebook

### PR DESCRIPTION
**Pull request by mrkumar9001**
**REQUIRED:** Add a summary of your PR here, typically including why the change is needed and what was changed. Include any design alternatives for discussion purposes.

<br>
feat: add fetch historical feature values notebook
<br><br><br>

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [ ] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [ ] Follow the style and grammar rules outlined in the above notebook template.
- [ ] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [ ] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [ ] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

<br>

2. If you are opening a PR for `Community Notebooks` under the [notebooks/community](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/community) folder:
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/community/CODEOWNERS) file under the `Community Notebooks` section, pointing to the author or the author's team.
- [ ] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).

<br>

3. If you are opening a PR for `Community Content` under the [community-content](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/community-content) folder:
- [ ] Make sure your main `Content Directory Name` is descriptive, informative, and includes some of the key products and attributes of your content, so that it is differentiable from other content
- [ ] The main content directory has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/community-content/CODEOWNERS) file under the `Community Content` section, pointing to the author or the author's team.
- [ ] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).

### CODE REVIEW

File: notebooks/official/feature_store/offline_feature_serving_from_bigquery_with_feature_registry.ipynb [hunk 1]
```diff
@@ -0,0 +1,472 @@
...
+        "fg: FeatureGroup = FeatureGroup.create(\n",
+        "    f\"{FEATURE_GROUP_ID}\",\n",
+        "    fs_utils.FeatureGroupBigQuerySource(\n",
+        "        uri=f\"bq://{BQ_TABLE_URI}\", entity_id_columns=[\"users\"]\n",
+        "    ),\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "code",
```
Comment (sen-sam) [inline notebooks/official/feature_store/offline_feature_serving_from_bigquery_with_feature_registry.ipynb:301]: Add a markdown block before this code block to describe the outcome of the code. For example: "Create the feature `movie_feature` within in the feature group and associate it with the column `movies` in the BigQuery source table."

File: notebooks/official/feature_store/offline_feature_serving_from_bigquery_with_feature_registry.ipynb [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+        "movie_feature: Feature = fg.create_feature(\"movies\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "c71964219962"
+      },
+      "source": [
+        "### Fetch with same entity at multiple timestamps"
```
Comment (sen-sam) [inline notebooks/official/feature_store/offline_feature_serving_from_bigquery_with_feature_registry.ipynb:317]: I feel this should be a H2, which would align the structure of the tutorial with the objectives.

Also, suggested edit: "Fetch historical feature values from the feature data source" or simply "Fetch historical feature values".

Provide a brief description of the subsections here. For example, "Fetch feature values at different timestamps by specifying one or more entities."

File: notebooks/official/feature_store/offline_feature_serving_from_bigquery_with_feature_registry.ipynb [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+      ]
+    },
+    {
```
Comment (sen-sam) [inline notebooks/official/feature_store/offline_feature_serving_from_bigquery_with_feature_registry.ipynb:320]: Add a description of the code before this block. For example, "Fetch feature values for the same entity at multiple timestamps."

Comment (sen-sam) [inline notebooks/official/feature_store/offline_feature_serving_from_bigquery_with_feature_registry.ipynb:320]: Add a H3 here: "Fetch historical feature values for an entity".

File: notebooks/official/feature_store/offline_feature_serving_from_bigquery_with_feature_registry.ipynb [hunk 4]
```diff
@@ -0,0 +1,472 @@
...
+        ")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "f9ecae3005df"
+      },
+      "source": [
+        "### Fetch with different entities"
```
Comment (sen-sam) [inline notebooks/official/feature_store/offline_feature_serving_from_bigquery_with_feature_registry.ipynb:350]: Suggested edit for H3: "Fetch historical feature values for multiple entities"

Add a description here, similar to the preceding H3: "Fetch feature values for multiple entities and timestamps. <Mention whether both the timestamps are considered for each entity.>"

**Review by sen-sam — COMMENTED** [review]
Suggested a few edits. Overall, it looks good. The code blocks should ideally have descriptions associated with them.

**Comment by gericdong** [comment]
@mrkumar9001: please let me know if this PR is ready. Thanks.

**Review by gericdong — APPROVED** [review]

**Pull request merged**